pepper-apis/src/main/java/org/broadinstitute/ddp/event/pubsubtask/impl/updateprofile/UpdateEmailHandler.java

```diff
@@ -0,0 +1,86 @@
+package org.broadinstitute.ddp.event.pubsubtask.impl.updateprofile;
+
+import static org.broadinstitute.ddp.event.pubsubtask.api.PubSubTaskLogUtil.infoMsg;
+import static org.slf4j.LoggerFactory.getLogger;
+
+
+import com.auth0.client.mgmt.ManagementAPI;
+import org.broadinstitute.ddp.db.TransactionWrapper;
+import org.broadinstitute.ddp.db.TransactionWrapper.DB;
+import org.broadinstitute.ddp.db.dao.DataExportDao;
+import org.broadinstitute.ddp.db.dao.JdbiUser;
+import org.broadinstitute.ddp.db.dto.UserDto;
+import org.broadinstitute.ddp.event.pubsubtask.api.PubSubTask;
+import org.broadinstitute.ddp.event.pubsubtask.api.PubSubTaskException;
+import org.broadinstitute.ddp.util.Auth0Util;
+import org.jdbi.v3.core.Handle;
+import org.slf4j.Logger;
+
+public class UpdateEmailHandler {
+
+    private static final Logger LOG = getLogger(UpdateEmailHandler.class);
+
+    private static final String FIELD_EMAIL = "email";
+
+    public void updateEmail(String userGuid, PubSubTask.PayloadMap payloadMap) {
+        TransactionWrapper.useTxn(DB.APIS, handle -> updateEmail(handle, userGuid, payloadMap));
+    }
+
+    private void updateEmail(Handle handle, String userGuid, PubSubTask.PayloadMap payloadMap) {
+        if (payloadMap.getMap().containsKey(FIELD_EMAIL)) {
+            String email = payloadMap.getMap().get(FIELD_EMAIL);
+            var userDto = handle.attach(JdbiUser.class).findByUserGuid(userGuid);
+            if (userDto == null) {
+                throw new PubSubTaskException("User profile is not found for guid=" + userGuid);
+            }
+            validateUserForLoginDataUpdateEligibility(userDto);
+            LOG.info(infoMsg("Attempting to change the email of the user {}"), userGuid);
+            var mgmtAPI = Auth0Util.getManagementApiInstanceForUser(userDto.getUserGuid(), handle);
+            updateEmailInAuth0(handle, mgmtAPI, userDto, email, userGuid);
+        }
+    }
+
+    private void validateUserForLoginDataUpdateEligibility(UserDto userDto) {
+        String errMsg = null;
+        if (userDto == null) {
+            errMsg = "User " + userDto.getUserGuid() + " does not exist in Pepper";
+        }
+        if (userDto.getAuth0UserId() == null) {
+            errMsg = "User " + userDto.getUserGuid() + " is not associated with the Auth0 user " + userDto.getAuth0UserId();
+        }
+        if (errMsg != null) {
+            throw new PubSubTaskException(errMsg);
+        }
+    }
+
+    private void updateEmailInAuth0(Handle handle, ManagementAPI mgmtAPI, UserDto userDto, String email, String userGuid) {
+        var status = Auth0Util.updateUserEmail(mgmtAPI, userDto, email);
+        String errMsg = null;
+        switch (status.getAuth0Status()) {
+            case SUCCESS:
```

Comment:
I'm not sure we need to add more details in the message for a sub set of auth0 statuses.  Since we know how to find details in auth0, we could just put `status.getAuth0Status()` into the error message directly.

Reply:
This logic taken from UpdateUserEmailRoute.java. May be to make it a common part for both classes (UpdateUserEmailRoute.java and UpdateEmailHandler.java)? Actually you've already confirmed earlier that it needs to be done (this kind of refactoring). But I still not done it.